synapse/handlers/federation.py

```diff
@@ -2714,9 +2714,11 @@ async def _update_auth_events_and_context_for_auth(
                             event.event_id,
                             e.event_id,
                         )
-                        context = await self.state_handler.compute_event_context(e)
+                        missing_auth_event_context = (
+                            await self.state_handler.compute_event_context(e)
+                        )
                         await self._auth_and_persist_event(
-                            origin, e, context, auth_events=auth
+                            origin, e, missing_auth_event_context, auth_events=auth
```

Comment:
This will merged separately in https://github.com/matrix-org/synapse/pull/10439 but need the fix here as well